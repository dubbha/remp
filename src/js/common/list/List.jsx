import React from 'react';
import PropTypes from 'prop-types';
import EmptyList from './EmptyList';
import ListItem from './ListItem';
import Spinner from './Spinner';
import ErrorBoundary from '../errorBoundary';
import filmPropShape from '../utils/propShapes';
import './style.sass';

const filtered = (list, film) => {
  if (film) {
    return list.filter(item => item.id !== film.id);
  }
  return list;
};

const List = ({ results, film, onSelectFilm, isLoading }) => (
  <section className="list">
    <ErrorBoundary>
      { isLoading && <Spinner /> }
      { !isLoading && results && (
        results.length > 0
          ? filtered(results, film).map(item => (
            <ListItem
              item={item}
              key={item.id}
              onSelectFilm={onSelectFilm}
            />))
          : <EmptyList />
      )}
    </ErrorBoundary>
  </section>
);

List.defaultProps = {
  results: [],
  film: {},
};

List.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape(filmPropShape)),
  film: PropTypes.shape(filmPropShape),
  onSelectFilm: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default List;
