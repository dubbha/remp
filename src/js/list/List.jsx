import React from 'react';

const List = () => (
  <section>
    List
  </section>
);

export default List;
