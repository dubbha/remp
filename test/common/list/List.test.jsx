import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from 'common/list';

describe('List', () => {
  let props;

  beforeEach(() => {
    props = {
      results: [
        { id: 87654321 },
        { id: 13572468 },
        { id: 12345678 },
      ],
      film: {
        id: 12345678,
        title: 'TITLE',
        director: 'DIRECTOR',
        overview: 'OVERVIEW',
        poster_path: 'POSTER_PATH',
        release_date: '2000/12/31',
        runtime: 120,
        vote_average: 5.0,
        genre_ids: [1, 2, 3],
        cast: [
          { name: 'NAME1' },
          { name: 'NAME2' },
          { name: 'NAME3' },
        ],
      },
      onSelectFilm: jest.fn(),
      isLoading: false,
    };
  });

  it('should render a filtered list if passed with a film object', () => {
    const wrapper = shallow(<List {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a full list if not passed with a film object', () => {
    props.film = null;

    const wrapper = shallow(<List {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render an empty list notification if the list is empty', () => {
    props.results = [];

    const wrapper = shallow(<List {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a spinner if is currently loading', () => {
    const wrapper = shallow(<List {...props} isLoading />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
