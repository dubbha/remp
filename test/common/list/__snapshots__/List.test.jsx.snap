// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render a filtered list if passed with a film object 1`] = `
<section
  className="list"
>
  <ErrorBoundary>
    <ListItem
      item={
        Object {
          "id": 87654321,
        }
      }
      key="87654321"
      onSelectFilm={[Function]}
    />
    <ListItem
      item={
        Object {
          "id": 13572468,
        }
      }
      key="13572468"
      onSelectFilm={[Function]}
    />
  </ErrorBoundary>
</section>
`;

exports[`List should render a full list if not passed with a film object 1`] = `
<section
  className="list"
>
  <ErrorBoundary>
    <ListItem
      item={
        Object {
          "id": 87654321,
        }
      }
      key="87654321"
      onSelectFilm={[Function]}
    />
    <ListItem
      item={
        Object {
          "id": 13572468,
        }
      }
      key="13572468"
      onSelectFilm={[Function]}
    />
    <ListItem
      item={
        Object {
          "id": 12345678,
        }
      }
      key="12345678"
      onSelectFilm={[Function]}
    />
  </ErrorBoundary>
</section>
`;

exports[`List should render a spinner if is currently loading 1`] = `
<section
  className="list"
>
  <ErrorBoundary>
    <Spinner />
  </ErrorBoundary>
</section>
`;

exports[`List should render an empty list notification if the list is empty 1`] = `
<section
  className="list"
>
  <ErrorBoundary>
    <EmptyList />
  </ErrorBoundary>
</section>
`;
